cf-custom-resources/lib/wkld-custom-domain.js

```diff
@@ -0,0 +1,554 @@
+ * @param {string} responseStatus the response status, either 'SUCCESS' or 'FAILED'
+ * @param {string} physicalResourceId CloudFormation physical resource ID
+ * @param {object} [responseData] arbitrary response data object
+ * @param {string} [reason] reason for failure, if any, to convey to the user
+ * @returns {Promise} Promise that is resolved on success, or rejected on connection error or HTTP error response
+ */
+function report(
+  event,
+  context,
+  responseStatus,
+  physicalResourceId,
+  responseData,
+  reason
+) {
+  return new Promise((resolve, reject) => {
+    const https = require("https");
+    const { URL } = require("url");
+
+    let reasonWithLogInfo = `${reason} (Log: ${context.logGroupName}/${context.logStreamName})`;
+    let responseBody = JSON.stringify({
+      Status: responseStatus,
+      Reason: reasonWithLogInfo,
+      PhysicalResourceId: physicalResourceId || context.logStreamName,
+      StackId: event.StackId,
+      RequestId: event.RequestId,
+      LogicalResourceId: event.LogicalResourceId,
+      Data: responseData,
+    });
+
+    const parsedUrl = new URL(event.ResponseURL);
+    const options = {
+      hostname: parsedUrl.hostname,
+      port: 443,
+      path: parsedUrl.pathname + parsedUrl.search,
+      method: "PUT",
+      headers: {
+        "Content-Type": "",
+        "Content-Length": responseBody.length,
+      },
+    };
+
+    https
+      .request(options)
+      .on("error", reject)
+      .on("response", (res) => {
+        res.resume();
+        if (res.statusCode >= 400) {
+          reject(new Error(`Error ${res.statusCode}: ${res.statusMessage}`));
+        } else {
+          resolve();
+        }
+      })
+      .end(responseBody, "utf8");
+  });
+}
+
+exports.handler = async function (event, context) {
+  // Destruct resource properties into local variables.
+  const props = event.ResourceProperties;
+  let {
+    PublicAccessDNS: publicAccessDNS,
+    PublicAccessHostedZoneID: publicAccessHostedZoneID,
+  } = props;
+  const aliases = new Set(props.Aliases);
+
+  // Initialize global variables.
+  envHostedZoneID = props.EnvHostedZoneId;
+  envName = props.EnvName;
+  appName = props.AppName;
+  serviceName = props.ServiceName;
+  domainName = props.DomainName;
+  rootDNSRole = props.RootDNSRole;
+  certificateDomain = `${serviceName}-nlb.${envName}.${appName}.${domainName}`;
+  domainTypes = {
+    EnvDomainZone: {
+      regex: new RegExp(`^([^\.]+\.)?${envName}.${appName}.${domainName}`),
+      domain: `${envName}.${appName}.${domainName}`,
+    },
+    AppDomainZone: {
+      regex: new RegExp(`^([^\.]+\.)?${appName}.${domainName}`),
+      domain: `${appName}.${domainName}`,
+    },
+    RootDomainZone: {
+      regex: new RegExp(`^([^\.]+\.)?${domainName}`),
+      domain: `${domainName}`,
+    },
+  };
+
+  const physicalResourceID = event.LogicalResourceId; // The PhysicalResourceID never changes because LogicalResourceId never changes.
```

Comment:
(again ik this is not you) 
```suggestion
  const physicalResourceID = event.LogicalResourceId; // The PhysicalResourceID never changes because LogicalResourceId never changes. Therefore, a "Replacement" should never happen.
```
so that it's more obvious to others what this comment is trying to say 🤣